Dockerfile

```diff
@@ -1,84 +1,105 @@
-RUN gcloud components install gke-gcloud-auth-plugin
-
-# Install Krew and add its installation directory to PATH
-RUN sh -c "\
-    set -x; cd \$(mktemp -d) && \
-    OS=\$(uname | tr '[:upper:]' '[:lower:]') && \
-    ARCH=\$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\\(arm\\)\\(64\\)\\?.*/\\1\\2/' -e 's/aarch64$/arm64/') && \
-    KREW=krew-\${OS}_\${ARCH} && \
-    curl -fsSLO \"https://github.com/kubernetes-sigs/krew/releases/latest/download/\${KREW}.tar.gz\" && \
-    tar zxvf \"\${KREW}.tar.gz\" && \
-    ./\"\${KREW}\" install krew \
-    "
-
-# Add Krew to PATH
-ENV PATH="/root/.krew/bin:$PATH"
 
-# Install kube-lineage via Krew
-RUN kubectl krew install lineage
-
-# Copy the poetry configuration files into the container at /app
-COPY pyproject.toml poetry.lock* /app/
+# Set the architecture-specific kube lineage URLs
+ARG ARM_URL=https://github.com/Avi-Robusta/kube-lineage/releases/download/v2.0.1/kube-lineage-macos-latest-v2.0.1
+ARG AMD_URL=https://github.com/Avi-Robusta/kube-lineage/releases/download/v2.0.1/kube-lineage-ubuntu-latest-v2.0.1
+# Define a build argument to identify the platform
+ARG TARGETPLATFORM
+# Conditional download based on the platform
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
+        curl -L -o kube-lineage $ARM_URL; \
+    elif [ "$TARGETPLATFORM" = "linux/amd64" ]; then \
+        curl -L -o kube-lineage $AMD_URL; \
+    else \
+        echo "Unsupported platform: $TARGETPLATFORM"; exit 1; \
+    fi
+RUN chmod 777 kube-lineage
+RUN ./kube-lineage --version
 
+# Set up poetry
 ARG PRIVATE_PACKAGE_REGISTRY="none"
 RUN if [ "${PRIVATE_PACKAGE_REGISTRY}" != "none" ]; then \
     pip config set global.index-url "${PRIVATE_PACKAGE_REGISTRY}"; \
     fi \
-    && pip install poetry     
-
-# Increase poetry timeout in case package registry times out
+    && pip install poetry    
 ARG POETRY_REQUESTS_TIMEOUT
-RUN poetry config virtualenvs.create false \
-    && if [ "${PRIVATE_PACKAGE_REGISTRY}" != "none" ]; then \
+RUN poetry config virtualenvs.create false
+COPY pyproject.toml poetry.lock /app/
+RUN if [ "${PRIVATE_PACKAGE_REGISTRY}" != "none" ]; then \
     poetry source add --priority=primary artifactory "${PRIVATE_PACKAGE_REGISTRY}"; \
     fi \
     && poetry install --no-interaction --no-ansi --no-root
 
-#COPY config.yaml /app/
+# Final stage
+FROM python:3.11-slim
+
+ENV PYTHONUNBUFFERED=1
+ENV PATH="/venv/bin:$PATH"
+ENV PYTHONPATH=$PYTHONPATH:.:/app/holmes
+
+WORKDIR /app
 
+COPY --from=builder /app/venv /venv
 COPY . /app
 
+
+RUN apt-get update \
+    && apt-get install -y \
+       git \
+       apt-transport-https \
+       gnupg2
+
+# Set up kubectl
+COPY --from=builder /app/Release.key Release.key
+RUN cat Release.key |  gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
+    && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
+    && apt-get update
```

Comment:
why do we need `apt-get update` at the end?

Reply:
we just added the kubernetes repo to apt 
`tee /etc/apt/sources.list.d/kubernetes.list`
so we needed update the package index to install kubectl